pkg/db/topology.go

```diff
@@ -193,28 +181,14 @@ func DeleteSystemTemplate(systemTemplate *v1.SystemTemplate) error {
 	if tx.Error != nil {
 		return tx.Error
 	}
-	systemID, err := DeleteSystem(systemTemplate.GetPersistedID(), deleteTime)
-	if err != nil {
-		return err
-	}
-	return DeleteComponnents(systemID, deleteTime)
-}
-
-func DeleteSystem(systemTemplateID string, deleteTime time.Time) (string, error) {
-	logger.Infof("Deleting system associated with template: %s", systemTemplateID)
-	systemModel := &pkg.System{}
-	tx := Gorm.Find(systemModel).Where("system_template_id = ?", systemTemplateID).UpdateColumn("deleted_at", deleteTime)
-	if tx.Error != nil {
-		return "", tx.Error
-	}
-	return systemModel.ID.String(), nil
+	return DeleteComponnents(systemTemplate.GetPersistedID(), deleteTime)
 }
 
 // DeleteComponents deletes all components associated with a system
 func DeleteComponnents(systemID string, deleteTime time.Time) error {
```

Comment:
A system template might generate many components - we would want to delete by path
```suggestion
func DeleteComponents(componentId string, deleteTime time.Time) error {
```

Reply:
@moshloop 

This function is called when a systemTemplate object is deleted... In that case, all the components need to be deleted that were linked to that systemTemplateID... For the components that stops existing....i.e. a pod got  deleted in the cluster... it is handled in the difference as part of the sync job.